app-info/pwa-metainfo-generator.py

```diff
@@ -144,6 +144,11 @@ def copy_metainfo_from_manifest(url, app_component, manifest, categories,
             image_element.set('height', size.split('x')[1])
             if 'label' in screenshot:
                 ET.SubElement(screenshot_element, 'caption').text = screenshot['label']
+    elif old_component is not None:
+        # If the existing component has screenshots, retain them.
+        screenshots_element = old_component.find("screenshots")
+        if screenshots_element:
+            app_component.append(screenshots_element)
```

Comment:
I think we may later want to do this for more elements, e.g. for added summaries (see your later change) & descriptions at least.

Reply:
Right. I figured for simpler things like the one line summary override, the CSV was fine. For the description, I think we would want something like that.

Reply:
After the changes in #33, I decided to change course here in order to maintain the newly added fields. Rather than editing the XML manually and trying to retain the customizations, I decided to switch the input format to YAML so it could express the intent directly. Particularly for screenshots and descriptions where you really need structured data rather than trying to mash it into a single CSV field.

Reply:
Perhaps the input format should be a single URL, just used to fill out the metainfo boilerplate once? I don't see much value in mashing e.g. OARS into a CSV or YAML format when there's a perfectly good XML format right there. Then there could be a refresh operation which takes the XML and updates it.

Reply:
Maybe you're right. I went ahead and hacked through the YAML handling, but I do see your point. The nice thing is that's repeatable. Let me know what you think.